doc/topics/development/contributing.rst

```diff
@@ -4,481 +4,604 @@
+Because of its widespread use, there are many resources for learning
+more about Git. One popular resource is the free online book `Learn Git
+in a Month of
+Lunches <https://www.manning.com/books/learn-git-in-a-month-of-lunches>`__.
 
+pyenv, Virtual Environments, and You
+----------------------------------------
 
-Quickstart
-----------
+We recommend `pyenv <https://github.com/pyenv/pyenv>`__, since it allows
+installing multiple different Python versions, which is important for
+testing Salt across all the versions of Python that we support.
 
-If you just want to get started before reading the rest of this guide, you can
-get the process started by running the following:
+On Linux
+~~~~~~~~
 
-.. code-block:: bash
+Install pyenv:
 
-    python3 -m pip install --user pre-commit
-    git clone --origin upstream https://github.com/saltstack/salt.git
-    cd salt
-    pre-commit install
+::
 
-While those commands are running, finish reading the rest of this guide.
+   git clone https://github.com/pyenv/pyenv.git ~/.pyenv
+   export PATH="$HOME/.pyenv/bin:$PATH"
+   git clone https://github.com/pyenv/pyenv-virtualenv.git $(pyenv root)/plugins/pyenv-virtualenv
 
+On Mac
+~~~~~~
 
-Pre-commit
-----------
+Install pyenv using brew:
 
-To reduce friction during the development process, SaltStack uses `pre-commit
-<pre-commit_>`_. This tool adds pre-commit hooks to git to automate several
-processes that used to be manual. Rather than having to remember to run several
-different tools before you commit, you only have to run ``git commit``, and you
-will be notified about style and lint issues before you ever open a PR.
+::
 
-.. warning::
-    Currently there is an issue with the pip-tools-compile pre-commit hook on windows.
-    The details around this issue are included here:
-    https://github.com/saltstack/salt/issues/56642.
-    Please ensure you export ``SKIP=pip-tools-compile`` to skip pip-tools-compile.
+   brew update
+   brew install pyenv
+   brew install pyenv-virtualenv
 
-Salt Coding Style
------------------
+--------------
 
-After the 3000 release, SaltStack is `joining the ranks <SEP 15_>`_ of projects
-in adopting the `Black code formatter <Black_>`_ in order to ease the adoption
-of a unified code formatting style.
+Now add pyenv to your ``.bashrc``:
 
-Where Black is silent, SaltStack has its own coding style guide that informs
-contributors on various style points. Please review the :ref:`Salt Coding Style
-<coding-style>` documentation for information about Salt's particular coding
-patterns.
+::
 
-Within the :ref:`Salt Coding Style <coding-style>` documentation, there is a
-section about running Salt's ``.testing.pylintrc`` file. SaltStack recommends
-running the ``.testing.pylintrc`` file on any files you are changing with your
-code contribution before submitting a pull request to Salt's repository.
+   export PATH="$HOME/.pyenv/bin:$PATH" >> ~/.bashrc
```

Comment:
```suggestion
   echo 'export PATH="$HOME/.pyenv/bin:$PATH"' >> ~/.bashrc
```

Reply:
Nice catch! I thought I fixed that but apparently I missed it again :weary: